From dsd@gentoo.org Sat Nov 12 09:54:01 2005
Message-ID: <43762BCA.7070909@gentoo.org>
Date: Sat, 12 Nov 2005 17:52:10 +0000
From: Daniel Drake <dsd@gentoo.org>
To: Greg KH <greg@kroah.com>
CC: USB Storage list <usb-storage@lists.one-eyed-alien.net>, Matthew Dharm <mdharm-usb@one-eyed-alien.net>, <pfavr@how.dk>
Subject: [PATCH] usb-storage: Fix detection of kodak flash readers in shuttle_usbat driver

Peter Favrholdt reported that his Kodak flash device was getting
detected as a CDROM, and he helped me track this down to the fact that
the device takes a long time (approx 440ms!) to reset.

This patch increases the delay to 500ms, which solves the problem.

Signed-off-by: Daniel Drake <dsd@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/storage/shuttle_usbat.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/storage/shuttle_usbat.c
+++ gregkh-2.6/drivers/usb/storage/shuttle_usbat.c
@@ -853,7 +853,7 @@ static int usbat_identify_device(struct 
 	rc = usbat_device_reset(us);
 	if (rc != USB_STOR_TRANSPORT_GOOD)
 		return rc;
-	msleep(25);
+	msleep(500);
 
 	/*
 	 * In attempt to distinguish between HP CDRW's and Flash readers, we now
